connectivity/FEATURE_BLE/source/cordio/source/GattServerImpl.cpp

```diff
@@ -978,8 +1002,8 @@ uint8_t GattServer::atts_read_cb(
     attsAttr_t *pAttr
 )
 {
-    GattCharacteristic *auth_char = getInstance().get_auth_char(handle);
-    if (auth_char && auth_char->isReadAuthorizationEnabled()) {
+    char_auth_callback *auth_cb = getInstance().get_auth_callback(handle);
+    if (auth_cb && auth_cb->read_cb) {
```

Comment:
I'm not a bit fan of pointer to bool coersion.
I'd prefer a 
```suggestion
    if ((auth_cb != NULL) && (auth_cb->read_cb != NULL)) {
```